string[] commands
bool go_home_on_finish
---
bool ok
string message
